other/majority_vote_algorithm.py

```diff
@@ -0,0 +1,28 @@
+import collections
+
+"""
+This is Booyer-Moore Majority Vote Algorithm. The problem statement goes like Given an integer array of size n, find all elements that appear more than ⌊ n/k ⌋ times.
+We have to solve in O(n) time and O(1) Space.
+URL : https://en.wikipedia.org/wiki/Boyer%E2%80%93Moore_majority_vote_algorithm
+
+"""
+
+
+def majority_element(nums: list[int], partition: int) -> list[int]:
```

Comment:
How does this relate to a real world political election with votes?

Are `nums` votes?  If not, what do `nums` represent?

Is `partition` the percent that of the votes that a candidate needs to win the election?  If not, what does `partition` represent?

Overall the variable names do not explain what the variables do.
`n` is 100% unclear.  Single letter variable names were supercool in the 1970's but not so much today.
`ctr` is a counter but what is it counting?  Is it a `vote_counter`, `candidate_counter`, `elector_counter`?

Reply:
OK...  Having read thru the wiki description, my advice would do the following renames (if you agree):
```
nums --> votes
n --> vote
ctr --> candidates
partition --> max_candidates
```
Two things jump out about this algorithm:
1. It is a streaming algorithm so votes should be an iterable, not a list.
2. It is trying to use a _constant number of words of memory_ which is rarely done today.

In your example, we can only keep track of the top 3 candidates at a time.  This is trivial because the `votes` only contain 3 different candidates.

`collections.Counter` is a powertool for this use case because it will solve the example with two lines:
```python
>>> from collections import Counter
>>> Counter([1, 2, 2, 3, 1, 3, 2]).most_common()
[(2, 3), (1, 2), (3, 2)]
```
Candidate number 2 wins with 3 votes while candidates 1 and 3 each got 2 votes.

This solution however is not streaming (all votes have already been tallied at the start of the process) and it is not constrained to a constant memory size (max_candidates is ignored).